src/app/components/dialog/dialog.ts

```diff
@@ -549,6 +551,7 @@ export class Dialog implements AfterContentInit, OnInit, OnDestroy {
     }
 
     ngOnInit() {
+        this.ariaLabelledBy = this.getAriaLabelledBy();
```

Comment:
The behavior is not the same, since the header can change and the string cannot be recalculated. You can write a pure pipe and it will be recalculated when the header changes.

Reply:
Thanks for check.
I couldn't replicate the error of the user, only I did this PR because was very similar as I wrote in the post.

In my opinion it's working perfectly and I cannot see the error but @psarno had the problem and was very similar to another that I did before.

If you have a better solution you can share here or send new commits to this PR, but it's difficult to check because I cannot replicate the error

Reply:
In this case, `header` is an `Input` property and can be changed at runtime, and `ariaLabelledBy` is evaluated only once in `OnInit`. This `pr` breaks the previous behavior. You can use for example `ngOnChanges` and remove in `ngOnInit` line.

```ts
public ngOnChanges({ header }: SimpleChanges): void {
  if (header) {
    this.ariaLabelledBy = this. getAriaLabelledBy();
  }
}
```

Reply:
Thanks again. I did a new commit with your suggestion.
I would like to replicate the problem but I can't, so I'm trusting in our work here haha.
Check again please! :)